### PR TITLE
Adds support for I18n translations

### PR DESCRIPTION
**Pull request by mylesmegyesi**
These changes add support for translating labels and error messages. Also splits up the fields.rb file into multiple files in order to avoid a 1000+ line file.

### CODE REVIEW

File: lib/bureaucrat.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
+$: << File.expand_path('..', __FILE__)
```
Comment (foca) [inline lib/bureaucrat.rb:1]: Never, EVER, do this. The $LOAD_PATH should not be altered by libraries, only by library managers (like RubyGems). If you want to locally have `./lib` in your load path, run ruby like this:

```
$ ruby -Ilib …
```

or

```
$ RUBYLIB=lib ruby …
```

If you don't want to type that every time, just `export RUBYLIB=lib` in your .bashrc or similar, and then ./lib will always be in your $LOAD_PATH.

File: lib/bureaucrat.rb [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+
+require 'i18n'
+
+root = File.expand_path('../..', __FILE__)
+I18n.load_path += Dir[File.join(root, 'locales', '**', '*.yml').to_s]
+
 module Bureaucrat
-  VERSION = '0.10.0'
+  VERSION = '0.11.3'
```
Comment (foca) [inline lib/bureaucrat.rb]: Don't change the version when submitting a pull request. Let the library's manager choose which version goes with changes.

**Comment by foca** [comment]
There's a lot going on in this pull request.

First, there's the DateField thing. That should be a pull-request on its own. @seanbrant could you please submit that in a pull request? :)

Same with @sittercity's fix. That should go in a different PR.

Then there's a few things:
- Adding bundler, which should not be used for gems.
- Adding a dependency on ActiveSupport just for the `String#underscore` method. Even if ActiveSupport 3.0 is modular and requiring one thing doesn't require the entire library, you're still requiring things that extend core ruby objects. That's _NOT_ nice. Monkeypatching core/stdlib classes is not something you should do. I personally wouldn't include ActiveSupport, but even then, I'd just require the inflector and do `Inflector.underscore(…)` instead of extending String with a bunch of methods.
- Then there's the mangling with the $LOAD_PATH and bumping the gem version that I already covered in previous comments.
- And finally, there's i18n support.

Personally, I'd remove bundler, ActiveSupport, and the $LOAD_PATH and VERSION changes before even looking at the rest of the PR.

**Comment by foca** [comment]
Er, I meant @dadarek's bug fix, not @sittercity.

**Comment by tizoc** [comment]
Hi @mylesmegyesi, thanks for the patches.

As @foca explained, I can't merge this in it's current state, there are too many different and unrelated things in this pull-request, and each should come as a separate pull request. There is also a lot of code restructuring which makes it very hard for me to undestand what was actually changed.

Having said that, I think i18n doesn't belong to the core of bureaucrat and would be better done as a separate gem (bureaucrat-i18n?)

**Pull request closed without merging**